Assets/Scripts/Singleton/TankController.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TankController : GenericSingleton<TankController>
+{
+    public Joystick joystick;
```

Comment:
Here Serialize field  that we can change in the inspector will be good not public one .
   [SerializeField] private Joystick joystick;